rest/shared/src/main/scala/io/udash/rest/internal/UsesREST.scala

```diff
@@ -0,0 +1,126 @@
+      annotations.exists(_.isInstanceOf[SkipRESTName])
+
+    def findRestName(annotations: Seq[MetadataAnnotation]): Option[String] =
+      annotations.find(_.isInstanceOf[RESTName]).map(_.asInstanceOf[RESTName].restName)
+
+    def parseInvocation(inv: framework.RawInvocation, metadata: RPCMetadata[_]): Unit = {
+      val rpcMethodName: String = inv.rpcName
+      val methodMetadata = metadata.signatures(rpcMethodName)
+
+      if (!shouldSkipRestName(methodMetadata.annotations))
+        urlBuilder += findRestName(methodMetadata.annotations).getOrElse(rpcMethodName)
+      methodMetadata.paramMetadata.zip(inv.argLists).foreach { case (params, values) =>
+        params.zip(values).foreach { case (param, value) =>
+          val paramName: String = findRestName(param.annotations).getOrElse(param.name)
+          val argTypeAnnotations = param.annotations.filter(_.isInstanceOf[ArgumentType])
+          if (argTypeAnnotations.size > 1) throw new RuntimeException(s"Too many parameter type annotations! ($argTypeAnnotations)")
+          argTypeAnnotations.headOption match {
+            case Some(_: Header) =>
+              headersArgsBuilder.+=((paramName, rawToHeaderArgument(value)))
+            case Some(_: Query) =>
+              queryArgsBuilder.+=((paramName, rawToQueryArgument(value)))
+            case Some(_: URLPart) =>
+              urlBuilder += rawToURLPart(value)
+            case Some(_: Body) =>
+              body = rawToString(value)
+            case _ => throw new RuntimeException(s"Missing `${param.name}` parameter type annotations! ($argTypeAnnotations)")
+          }
+        }
+      }
+    }
+
+    def findRestMethod(inv: framework.RawInvocation, metadata: RPCMetadata[_]): RESTConnector.HTTPMethod = {
+      val rpcMethodName: String = inv.rpcName
+      val methodMetadata = metadata.signatures(rpcMethodName)
+      val methodAnnotations = methodMetadata.annotations.filter(_.isInstanceOf[RESTMethod])
+      if (methodAnnotations.size > 1) throw new RuntimeException(s"Too many method type annotations! ($methodAnnotations)")
+      methodAnnotations.headOption match {
+        case Some(_: GET) => RESTConnector.GET
+        case Some(_: POST) => RESTConnector.POST
+        case Some(_: PATCH) => RESTConnector.PATCH
+        case Some(_: PUT) => RESTConnector.PUT
+        case Some(_: DELETE) => RESTConnector.DELETE
+        case _ => throw new RuntimeException(s"Missing method type annotations! ($methodAnnotations)")
+      }
+    }
+
+    var metadata: RPCMetadata[_] = rpcMetadata
+    getterChain.reverse.foreach(inv => {
+      parseInvocation(inv, metadata)
+      metadata = metadata.getterResults(inv.rpcName)
+    })
+    parseInvocation(invocation, metadata)
+
+    val restMethod: HTTPMethod = findRestMethod(invocation, metadata)
+
+    connector.send(s"/${urlBuilder.result().mkString("/")}", restMethod, queryArgsBuilder.result(), headersArgsBuilder.result(), body) onComplete {
```

Comment:
I'd inline restMethod and use named parameters here, this is probably the single most important line in this extension.